core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4292,18 +4294,27 @@ private static List<Expression> args(RexCall call,
       operand0 = operands.left;
       operand1 = operands.right;
       boolean nonPositiveIsNull = library == SqlLibrary.MYSQL ? true : false;
```

Comment:
We should pass the library as the Parameter. Then the code logic will be clear. WDYT?

Reply:
Maybe we should refactor the code in another issue. The current code here has poor extensibilit.

Reply:
Yes, I have to admit this. https://issues.apache.org/jira/browse/CALCITE-6550 is already working on this. I'll close it first.

Reply:
@NobiGo Could you please post your opinion at https://issues.apache.org/jira/browse/CALCITE-6550? I think it is a very complicated issue.

Reply:
I don't think you should pass in a library at run time. The runtime should not be worrying about high-level concepts like llibrary.